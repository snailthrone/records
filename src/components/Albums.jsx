import React from 'react';
import styled from 'styled-components';

const Albums = styled.div`
  font-size: 14px;
  line-height: 1.75;
  margin: auto;
  max-width: 600px;
  padding-bottom: 100px;
  width: 95%;
  @media screen and (min-width: 600px) {
    font-size: 18px;
    width: 100%;
  }
`;

const Heading = styled.h3`
  font-weight: 700;
  margin: auto auto 10px auto;
`;

const Album = styled.p`
	margin: auto auto 15px auto;
	text-decoration: ${props => props.bought ? 'line-through' : 'none'};
`;

const AlbumArtist = styled.span``;
const AlbumTitle = styled.span``;
const AlbumDate = styled.span``;

export default props => (
	<Albums>
	<Heading>Artist – Album (date)</Heading>
	{
		props.data.map((data, i) => {
			if (data.artist.toLowerCase().match(props.value) || data.album.toLowerCase().match(props.value) || data.date.match(props.value)) {
				return (
					<Album bought={data.bought} key={i}>{i+1}.
						<AlbumArtist className="artist"> {data.artist}</AlbumArtist> –
						<AlbumTitle className="album"> {data.album} </AlbumTitle>
						{(data.date && data.date != 'null') && <AlbumDate className="date">({data.date})</AlbumDate>}
					</Album>
				);
			}
		})
	}
	</Albums>
);