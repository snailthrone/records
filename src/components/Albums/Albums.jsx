import PropTypes from 'prop-types'
import React, { Fragment } from 'react'
import styled from 'styled-components'

const Container = styled.ul`
  list-style: none;
  margin: 10px auto auto auto;
  max-width: 600px;
  padding: 0 0 100px 0;
  width: 90%;
  @media screen and (min-width: 600px) {
    width: 100%;
  }
`

const Heading = styled.h3`
  font-family: 'Questrial', Verdana, sans-serif;
  font-size: 16px;
  font-weight: 700;
  margin: 0.8em auto;
  max-width: 600px;
  width: 90%;
  @media screen and (min-width: 600px) {
    width: 100%;
  }
`

const Album = styled.li`
  font-family: 'Questrial', Verdana, sans-serif;
  font-size: 12px;
  line-height: 1.35em;
  margin: auto auto .4em auto;
  text-decoration: ${({ bought }) => (bought ? 'line-through' : 'none')};
  @media screen and (min-width: 600px) {
    font-size: 14px;
  }
`

const Albums = ({ data, value }) => (
  <Fragment>
    <Heading>
      Artist – Album (date)
    </Heading>
    <Container>
      {
        data.map(({
          artist, album, bought, date,
        }, i) => {
          if (artist.toLowerCase().match(value.toLowerCase())
              || album.toLowerCase().match(value.toLowerCase())
              || (date && date.match(value.toLowerCase()))
          ) {
            return (
              <Album bought={bought} key={`album-'${(i + 1)}`}>
                {
                  date ? (
                    `${artist} – ${album} (${date})`
                  ) : (
                    `${artist} – ${album}`
                  )
                }
              </Album>
            )
          }
          return false
        })
      }
    </Container>
  </Fragment>
)

Albums.propTypes = {
  data: PropTypes.arrayOf(() => {}).isRequired,
  value: PropTypes.string,
}

Albums.defaultProps = {
  value: '',
}

export default Albums
