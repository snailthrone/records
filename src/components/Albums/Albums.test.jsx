import React from 'react'
import renderer from 'react-test-renderer'

import Albums from './Albums'

const sample = [{
  artist: 'Deafheaven',
  album: 'Ordinary Corrupt Human Love',
  date: '13.7.2018',
  bought: true,
}, {
  artist: 'Urfaust',
  album: 'Constellatory Practice',
  date: '25.5.2018',
  bought: false,
}]

it('Renders corretly', () => {
  const tree = renderer.create(
    <Albums
      data={sample}
    />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Renders corretly if value doesn\'t match', () => {
  const tree = renderer.create(
    <Albums
      data={sample}
      value="asdasdasdasd"
    />,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
