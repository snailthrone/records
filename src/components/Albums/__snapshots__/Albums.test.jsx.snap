// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders corretly 1`] = `
Array [
  <h3
    className="sc-bwzfXH bUqtjj"
  >
    Artist – Album (date)
  </h3>,
  <ul
    className="sc-bdVaJa euFgJ"
  >
    <li
      className="sc-htpNat flGfQk"
    >
      Deafheaven – Ordinary Corrupt Human Love (13.7.2018)
    </li>
    <li
      className="sc-htpNat jHhbwj"
    >
      Urfaust – Constellatory Practice
    </li>
  </ul>,
]
`;

exports[`Renders corretly if value doesn't match 1`] = `
Array [
  <h3
    className="sc-bwzfXH bUqtjj"
  >
    Artist – Album (date)
  </h3>,
  <ul
    className="sc-bdVaJa euFgJ"
  />,
]
`;
